lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp

```diff
@@ -2365,11 +2368,70 @@ CompilerType TypeSystemSwiftTypeRef::GetChildCompilerTypeAtIndex(
 }
 
 size_t TypeSystemSwiftTypeRef::GetIndexOfChildMemberWithName(
-    opaque_compiler_type_t type, const char *name, bool omit_empty_base_classes,
-    std::vector<uint32_t> &child_indexes) {
+    opaque_compiler_type_t type, const char *name, ExecutionContext *exe_ctx,
+    bool omit_empty_base_classes, std::vector<uint32_t> &child_indexes) {
+  if (auto *exe_scope = exe_ctx->GetBestExecutionContextScope())
+    if (auto *runtime =
+            SwiftLanguageRuntime::Get(exe_scope->CalculateProcess()))
+      if (auto index_size = runtime->GetIndexOfChildMemberWithName(
+              GetCanonicalType(type), name, exe_ctx, omit_empty_base_classes,
+              child_indexes)) {
+#ifndef NDEBUG
+        // This block is a custom VALIDATE_AND_RETURN implementation to support
+        // checking the return value, plus the by-ref `child_indexes`.
+        if (!m_swift_ast_context)
+          return *index_size;
+        auto ast_type = ReconstructType(type);
+        if (!ast_type)
+          return *index_size;
+        std::vector<uint32_t> ast_child_indexes;
+        auto ast_index_size = m_swift_ast_context->GetIndexOfChildMemberWithName(
+                ast_type, name, exe_ctx, omit_empty_base_classes,
+                ast_child_indexes);
+        // The runtime has more info than the AST. No useful validation can be
+        // done.
+        if (*index_size > ast_index_size)
+          return *index_size;
+
+        auto fail = [&]() {
+          auto join = [](const auto &v) {
+            std::ostringstream buf;
```

Comment:
... by calling llvm::join from StringExtras.h http://llvm.org/doxygen/namespacellvm.html#aa1fd1719afe5b07ebe55f1d105c98916

Reply:
I went to use `llvm::join`, but it's for containers of strings, and this case is a `vector<int>` basically. Since I'd need to convert to strings, it was shorter to do this, and since it will hopefully soon be deleted (that's the plan for the validation code right?), I didn't force the issue of using `llvm::join`.